### ISSUE TITLE
RFE: Add abicheck tests

### ISSUE THREAD

**Issue by sgallagher** [rpminspect/rpminspect#144]
Before Taskotron was turned off, packages were able to gate on [`dist.abicheck`](https://pagure.io/task-abicheck) powered by libabigail and abidiff/abipkgdiff. This was really handy, as it helped avoid unexpected ABI breakage during updates.

It would be *really* nice if this functionality could move to rpminspect.

CC @dodji and @sinnykumari

**Comment by dcantrell** [comment]
I'm about to make the 0.13 release and will include this in the next release.  The abicheck in rpmdiff also handled the ABI guarantee levels that are now defined in the rpminspect-data package.  So I want to incorporate that as well so that Fedora has the ability to define how much we care about ABI guarantees on a per package basis if we want to.

**Comment by dcantrell** [comment]
Just an update on this.  I am actively working on this inspection for librpminspect.  The rpmdiff code contains a lot of stuff we don't really need to do anymore because libabigail covers what we care about.  This inspection also enforces the ABI compatibility levels as defined by the ABI compatibility whitelists in the rpminspect-data packages.

**Comment by dcantrell** [comment]
Another update.  I have added the abidiff inspection now which runs abidiff(1) to compare ELF shared objects across packages.  I have modified the peer matching code to handle library version numbers changing so you still get the ABI comparison run.  That also fixes up the unusual reporting of telling you that libfoo.so.1.1 was removed and libfoo.so.1.2 was added.  Now you don't see either of those warnings because rpminspect understands that the library has just been updated.  You do get the abidiff run for those pairs now.

What I still need to finish this week:

- Reading and checking against the ABI compatibility level file for the product release.  This is from rpmdiff and lists the different levels of ABI guarantees we give.  As defined, it is divided out by package name and only supports 4 levels.  I am going to implement it without an upper bound of ABI levels since beyond level 2 it's just grouping the reports together.  I also plan to implement support for individual library naming rather than just the packages.  It looks like support for that began but was not finished.
- Adding in a new inspection to run kmidiff on kernel modules.  This will basically be what I add for abidiff, but will only apply to kernel modules and will run kmidiff instead of abidiff.  I am breaking this out to a separate inspection so the user can enable or disable it if necessary.
- Integration test cases.
